src/js/client/common-dialogs-widget.js

```diff
@@ -0,0 +1,124 @@
+/* global Foundation */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.dialogs");
+
+/**
+ * DevPMT Base Dialog
+ *
+ * Creates the simplest dialog, currently based on the foundation
+ * modal dialog. Can be inherited to create more complex dialogs.
+ */
+fluid.defaults("gpii.devpmt.dialogs.baseDialog", {
+    gradeNames: ["gpii.devpmt.viewComponent"],
+    templates: {
+        initial: "common-dialog-widget"
+    },
+    model: {},
+    invokers: {
+        openDialog: {
+            funcName: "gpii.devpmt.dialogs.baseDialog.openDialog",
+            args: ["{that}", "{that}.dom.dialogContainer"]
+        },
+        closeDialog: {
+            funcName: "gpii.devpmt.dialogs.baseDialog.closeDialog",
+            args: ["{that}", "{that}.dom.dialogContainer"]
+        }
+    },
+    selectors: {
+        dialogContainer: "#pmt-dialog-container"
+    },
+    listeners: {
+        "onMarkupRendered.openDialog": {
+            func: "{that}.openDialog"
+        }
+    },
+    events: {
+        // Sometimes foundation can move around the markup for the
+        // dialog to float it, etc. and we can lose event bindings.
+        // This allows activity to happen immediately after the dialog
+        // is opened.
+        afterOpenDialog: null
+    }
+});
+
+gpii.devpmt.dialogs.baseDialog.openDialog = function (that, dialogContainer) {
+    /* eslint-disable no-new */
+    // This foundation plugin requires using the `new` operator to work.
+    new Foundation.Reveal(dialogContainer, {
+        appendTo: that.options.selectors.initial
+    });
+    /* eslint-enable no-new */
+    dialogContainer.on("closed.zf.reveal", function () {
+        that.destroy();
+    });
+    dialogContainer.foundation("open");
+    that.events.afterOpenDialog.fire();
+};
+
+gpii.devpmt.dialogs.baseDialog.closeDialog = function (that, dialogContainer) {
+    dialogContainer.foundation("close");
+};
+
+/**
+ * DevPMT Confirmation Dialog
+ *
+ * Suitable dialog for performing actions that require either an
+ * 'accept' or 'cancel' type answer. Such as adding a product to
+ * an NP Set, deleting a context from an NP Set, etc.
```

Comment:
NP Set -> preferences set. The term of "NP set" has been globally replaced in the universal repo. Please use new terms "preferences set" and "preferences safe" based on contexts. Thanks.

This comment applies to the entire pull request.

Reply:
all uses of npset have been removed!